server/src/main/java/org/opensearch/wlm/tracker/QueryGroupUsageHelper.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.wlm.tracker;
+
+import org.opensearch.cluster.metadata.QueryGroup;
+import org.opensearch.common.annotation.PublicApi;
+import org.opensearch.wlm.WorkloadManagementSettings;
+
+/**
+ * Utility class to provide utility methods at query group level
+ */
+@PublicApi(since = "2.18.0")
+public abstract class QueryGroupUsageHelper {
+    private WorkloadManagementSettings settings;
+
+    public WorkloadManagementSettings getSettings() {
+        return settings;
+    }
+
+    /**
+     * WorkloadManagementSettings setter
+     * @param settings
+     */
+    public void setSettings(WorkloadManagementSettings settings) {
+        this.settings = settings;
+    }
```

Comment:
Will be great if we can take this as constructor. Maybe we can avoid per `ResourceType` instantiation of `QueryGroupUsageHelper` by having node level threshold from WLM settings within `ResourceType`?

Reply:
I have pushed the `Function<WorkloadManagementSettings, Double>` as a member variable into the `ResourceType`